tasks/eslint.js

```diff
@@ -0,0 +1,42 @@
+import { CLIEngine } from 'eslint';
+
+const OPTION_DEFAULTS = {
+  paths: null,
+  cache: null,
+  fix: false
+};
+
+module.exports = grunt => {
+  grunt.registerMultiTask('eslint',function () {
+    const options = this.options(OPTION_DEFAULTS);
+
+    if (!options.paths) {
+      grunt.fatal(new Error('No eslint.options.paths specified'));
+      return;
+    }
+
+    const cli = new CLIEngine({
+      cache: options.cache,
+      fix: options.fix,
+      cwd: grunt.config.get('root'),
+    });
+
+    const report = cli.executeOnFiles(options.paths);
+
+    // output fixes to disk
+    if (options.fix) {
+      CLIEngine.outputFixes(report);
+    }
+
+    // log the formatted linting report
+    const formatter = cli.getFormatter();
+
+    const errTypes = [];
+    if (report.errorCount > 0) errTypes.push('errors');
+    if (report.warningCount > 0) errTypes.push('warning');
+    if (!errTypes.length) return;
+
+    grunt.log.write(formatter(report.results));
```

Comment:
Sounds good. How about renaming`report` to `errorReport` or adding a comment to make that clearer? Otherwise I can see how someone might have the same question I had from reading this code.